tests/run/t6534.scala

```diff
@@ -8,7 +8,7 @@ object Test {
   def main(args: Array[String]): Unit = {
     val b1 = new Bippy1(71)
     val b2 = new Bippy2(71)
-    assert(b1 == b1)
+    assert(b1 != b1)
```

Comment:
I think the previous `equals` behavior was unintentional and non-sensical. So I don't think we need to worry about migration, that should count as a bugfix. As far as I can tell, the boxing behavior is also the same as it was before. So I don't see a downside for doing this.

Boxing could be improved for value classes that define `equals`, but that's a separate issue for which @smarter would be the best person.

Reply:
> I think the previous equals behavior was unintentional and non-sensical. So I don't think we need to worry about migration, that should count as a bugfix.

Only making more programs compile is a safe bugfix, changing behavior of existing programs isn't safe. As you pointed out other times, they might depend on it without even knowing. We'd need to forbid the scenario in Scala 2 to make this safe.

And this behavior wasn't even accidental — you, Paul and Adriaan discussed it in https://github.com/scala/bug/issues/6534 and https://github.com/scala/scala/pull/1526, so it can't be _that_ nonsensical, and it sure isn't a bugfix.

That `-Xlint` warning for `class Bippy1(val x: Int) extends AnyVal with Foo { }` was even removed in
https://github.com/scala/scala/commit/1da48a45b62879c2bd2904342eeff7e6e568350a#diff-90c2f5c65d04e4b756b9989ff2e1b97dR339, but apparently only because there was no way to suppress it.